Add miqssh utilities to tools dir.

jeffmcutter opened the pull request:
These tools are very helpful for managing multiple appliances.
The miqgrep and miqtail commands are useful for single appliances as well when used with -r option to follow particular requests and associated tasks in the MIQ logs.
Additional details on the utilities are available in the README.md file.
See INSTALL.txt for setting this tool up for use the first time.

Fryguy commented:
@bdunne Please review.   I haven't reviewed yet, but I would be ok with merging not-perfect, and fixing anything up in follow up PRs

bdunne commented on tools/miqssh/miqhosts-gen:
```diff
@@ -0,0 +1,22 @@
+cd /var/www/miq/vmdb
+
+#echo 'select miq_servers.hostname, zones.name from miq_servers join zones on zones.id = miq_servers.zone_id;' | rails db | awk '$1!="hostname" && !/^---/ && !/^\(/ {print}' | sed -e 's/ //g' -e 's/|/ /' -e '/^$/d'
+psql -d vmdb_production -c 'select miq_servers.hostname, zones.name from miq_servers join zones on zones.id = miq_servers.zone_id;' | awk '$1!="hostname" && !/^---/ && !/^\(/ {print}' | sed -e 's/ //g' -e 's/|/ /' -e '/^$/d'
```
This could probably be simplified through rails.  What is the output you're looking for?

jeffmcutter replied:
I would like a better way to get this information.  There is a similar psql in miqssh that I don't like.  This one is getting a list of appliances along with the zone each appliance is in.

bdunne replied:
@jeffmcutter If you can supply what you want the output to look like, we can help craft the ruby to get the information for you.

bdunne commented on tools/miqssh/miqhosts.rb:
```diff
@@ -0,0 +1,59 @@
+#
+
+# Method load_file to read the hosts file and create useful datastructures.
+def load_file(file)
```
You probably want to check out [LinuxAdmin::Hosts](https://github.com/ManageIQ/linux_admin/blob/master/lib/linux_admin/hosts.rb)

jeffmcutter replied:
miqhosts has a mapping of appliance name to a csv separated list of groups.  miqhosts.rb searches miqhosts for all hosts that are in a particular group, and can list the groups.  LinuxAdmin::Hosts seems to be for manipulating the hostname and hosts files in the format of IP hostname and optional comment.  Can it handle the groups stuff?

Fryguy commented:
@jeffmcutter Can you share the video you made?

jeffmcutter commented:
@Fryguy here's the link to the video I made last week.  You may need to request access if you're outside.

https://drive.google.com/file/d/1w6IoWO_NQRsjFzCsV6WHq9OPi-L9PebU/view?usp=sharing

Bear in mind that since I made the video I renamed things from cf to miq and merged the command names with dashes into their parent and added the -r option instead.

simaishi commented:
Gaprindashvili backport details:

```
$ git log -1
commit 1cfedccba052f19620b450c6675b73ca2177e2aa
Author: Brandon Dunne <brandondunne@hotmail.com>
Date:   Wed Feb 7 14:00:48 2018 -0500

    Merge pull request #16926 from jeffmcutter/add_miqssh_tools
    
    Add miqssh utilities to tools dir.
    (cherry picked from commit 421fe97f25d2d837d305cdf3c263a566985eb13c)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1553362
```

simaishi commented:
Fine backport details:

```
$ git log -1
commit bad1222c1981d2f36d4e6c252932e2dff9ad3fff
Author: Brandon Dunne <brandondunne@hotmail.com>
Date:   Wed Feb 7 14:00:48 2018 -0500

    Merge pull request #16926 from jeffmcutter/add_miqssh_tools
    
    Add miqssh utilities to tools dir.
    (cherry picked from commit 421fe97f25d2d837d305cdf3c263a566985eb13c)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1553364
```
